sdk/core/switcher/entry.S

```diff
@@ -779,3 +779,77 @@ exception_entry_asm:
 #endif
 #endif // CONFIG_NO_SWITCHER_SAFETY
 	cret
+
+// Returns whether the trusted stack has space for N more calls.
+	.section .text, "ax", @progbits
+	.p2align 2
+	.type __Z23trusted_stack_has_spacei,@function
+__Z23trusted_stack_has_spacei:
+	li                 a2, TrustedStackFrame_size
+	mul                a2, a0, a2
+	cspecialr          ca0, mtdc
+	clhu               a1, TrustedStack_offset_frameoffset(ca0)
```

Comment:
```suggestion
    // Fetch offset from trusted stack; unless something has gone very wrong indeed,
    // this cannot trap.
	clhu               a1, TrustedStack_offset_frameoffset(ca0)
```

Reply:
That's defence in depth, but we are not relying on that.  We are relying on the fact that there is a single exit path that clears all registers and that if we trap in the switcher then we force unwind.